### PR TITLE
cosmic-edit: 0-unstable-2024-04-15 -> 1.0.0-alpha.1

### PR DESCRIPTION
**Pull request by ahoneybun**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by alyssais** [comment]
Same question as https://github.com/NixOS/nixpkgs/pull/332252#issuecomment-2271031763 — should we use the tag?

**Comment by ahoneybun** [comment]
> Same question as [#332252 (comment)](https://github.com/NixOS/nixpkgs/pull/332252#issuecomment-2271031763) — should we use the tag?

does that mean the version in package.nix should be epoch-1.0.0-alpha.1 ?

**Comment by alyssais** [comment]
> does that mean the version in package.nix should be epoch-1.0.0-alpha.1 ?

If we are using the tag, I think the version should be "1.0.0-alpha.1".  As I understand it, "epoch" is just the name for this incarnation of COSMIC, the one written in Rust and not part of GNOME, so I don't think that needs to be part of the version.  (But I could be wrong.)

**Comment by ahoneybun** [comment]
> > does that mean the version in package.nix should be epoch-1.0.0-alpha.1 ?
> 
> If we are using the tag, I think the version should be "1.0.0-alpha.1". As I understand it, "epoch" is just the name for this incarnation of COSMIC, the one written in Rust and not part of GNOME, so I don't think that needs to be part of the version. (But I could be wrong.)

if the version is the tag then it would include the epoch in there as well.

**Comment by alyssais** [comment]
I think it's the same situation as when a tag is prefixed with "v".  It's part of the tag, but we don't include it as part of the Nixpkgs version.

**Review by alyssais — CHANGES_REQUESTED** [review]
The tag should be in "rev", and I still don't think "epoch" is part of the version.  We almost always trim everything until the first number in Nixpkgs versions.

File: pkgs/by-name/co/cosmic-edit/package.nix [hunk 1]
```diff
@@ -21,12 +21,11 @@
 
 rustPlatform.buildRustPackage rec {
   pname = "cosmic-edit";
-  version = "unstable-2024-03-30";
-
```
Comment (alyssais) [inline pkgs/by-name/co/cosmic-edit/package.nix:25]: What happened to this line?  We usually leave a blank one between `version` and `src`, like there was before.

Comment (ahoneybun) [reply pkgs/by-name/co/cosmic-edit/package.nix:25]: Did it need to be blank? I see sometimes that it is there and sometimes no? It is in the same {} so I don't think it should be.

Comment (ahoneybun) [reply pkgs/by-name/co/cosmic-edit/package.nix:25]: Added back.

**Review by alyssais — APPROVED** [review]

**Pull request merged**